### PR TITLE
[ser] refacoring Broadcom SER test

### PR DESCRIPTION
**Pull request by yxieca**
Summary:
Refactoring SER test.

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### What is the motivation for this PR?
SER test is failing randomly on TD2/TH/TH2 platforms and consistently on TD3 platform.

#### How did you do it?
- Persist the memory information to a file for quick retest.
- Split skipping memory entries according to the ASIC type.
- Protect against truncated log entry parsing error.

Signed-off-by: Ying Xie <ying.xie@microsoft.com>

#### How did you verify/test it?
TD2/TD3/TH/TH2: with the fix passed multiple times basic and confident level test.

### CODE REVIEW

**Comment by yxieca** [comment]
retest vsimage please

File: tests/platform_tests/broadcom/files/ser_injector.py [hunk 1]
```diff
@@ -94,6 +247,31 @@ def run_cmd(cmd):
     stdout, stderr = out.communicate()
     return stdout, stderr
 
+def get_asic_type():
```
Comment (neethajohn) [inline tests/platform_tests/broadcom/files/ser_injector.py:250]: can you move this to a helper? might be useful for other tests as well.

Comment (yxieca) [reply tests/platform_tests/broadcom/files/ser_injector.py:250]: I can add one with another PR. but this is in the file executed on the DUT. It is standalone script.

Comment (yxieca) [reply tests/platform_tests/broadcom/files/ser_injector.py]: ok, scratch the another PR part. I added the helper function, also noticed that asic_type was used already. So I name the function get_asic_name() instead.

**Review by daall — APPROVED** [review]

**Pull request merged**